JASP-Engine/JASP/R/correlation.R

```diff
@@ -15,806 +15,721 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-Correlation <- function(dataset=NULL, options, perform="run", callback=function(...) 0, ...) {
-    if (is.null(dataset)) {
-        if (perform == "run") {
-            if (options$missingValues == "excludeListwise") {
-                dataset <- .readDataSetToEnd(columns.as.numeric=options$variables, exclude.na.listwise=options$variables)
-            } else {
-                dataset <- .readDataSetToEnd(columns.as.numeric=options$variables)
-			}
-		} else {
-			dataset <- .readDataSetHeader(columns.as.numeric=options$variables)
-		}
-    }
-    
-    results <- list()
-
-	meta <- list(
-	    list(name="title", type="title"),
-		list(name="correlations", type="table"),
-		list(name="plot", type="image"))
-
-	results[[".meta"]] <- meta
-
-	results[["title"]] <- "Correlation Matrix"
-	
-	state <- .retrieveState()
-	diff <- NULL
-
-	if (!is.null(state)) {
-	    diff <- .diff(options, state$options)
-	}
+Correlation <- function(dataset=NULL, options, perform="run", callback=function(...) 0, state=NULL, ...) {
+  
+  stateKey <- list(
+    correlationPlot = c("plotCorrelationMatrix", "plotDensities", "plotStatistics",
+                        "variables", "pearson", "kendallsTauB", "spearman",
+                        "confidenceIntervals", "confidenceIntervalsInterval", "hypothesis", "missingValues"),
+    tableData = c("hypothesis", "missingValues", "confidenceIntervalsInterval")
+  )
+
+  if (is.null(dataset)) {
+    if (perform == "run") {
+      if (options$missingValues == "excludeListwise") {
+        dataset <- .readDataSetToEnd(columns.as.numeric=options$variables, exclude.na.listwise=options$variables)
+      } else {
+        dataset <- .readDataSetToEnd(columns.as.numeric=options$variables)
+      }
+    } else {
+      dataset <- .readDataSetHeader(columns.as.numeric=options$variables)
+  	}
+  }
 	
-	correlation.plot <- NULL
-
-	if (perform == "init" & options$plotCorrelationMatrix) {
-	    if (!is.null(state) && !is.null(diff) && ((is.logical(diff) && diff == FALSE) || (is.list(diff) && (diff$confidenceIntervals == FALSE && diff$confidenceIntervalsInterval == FALSE
-			&& diff$hypothesis == FALSE && diff$kendallsTauB == FALSE && diff$missingValues == FALSE && diff$pearson == FALSE && diff$plotCorrelationMatrix == FALSE
-			&& diff$plotDensities == FALSE && diff$plotStatistics == FALSE && diff$spearman == FALSE && diff$variables == FALSE)))) {
-
-			# if only "Report significance", "Flag significant correlations", "Means and Standard Deviations" or "Cross-product deviations and covariances" have changed, the previous plot can be used
-
-			correlation.plot <- state$correlationPlot
-
-		} else {
-
-			variables <- unlist(options$variables)
-
-
-			if (length(variables) > 1) {
+	correlation.plot <- state$correlationPlot
+  stateTableData <- state$tableData
 
-				l <- length(variables)
+  if (options$plotCorrelationMatrix && is.null(correlation.plot)) {
```

Comment:
isTrue(options$plotCorrelationMatrix)

Reply:
`isTrue` is completely redundant. "plotCorrelationMatrix" speaks for itself!

Reply:
Yes I know, but there's no naming convention for our json input which is why it'll be good to specify it.